x-pack/test/security_solution_api_integration/test_suites/detections_response/rule_creation/create_rules.ts

```diff
@@ -30,43 +30,50 @@ import {
   getSimpleMlRule,
   getSimpleMlRuleOutput,
   waitForRuleSuccess,
-  getRuleForSignalTesting,
-  getRuleForSignalTestingWithTimestampOverride,
+  getRuleForAlertTesting,
+  getRuleForAlertTestingWithTimestampOverride,
   waitForAlertToComplete,
-  waitForSignalsToBePresent,
-  getThresholdRuleForSignalTesting,
+  waitForAlertsToBePresent,
+  getThresholdRuleForAlertTesting,
   waitForRulePartialFailure,
   createRule,
   deleteAllAlerts,
-} from '../../utils';
-import { createUserAndRole, deleteUserAndRole } from '../../../common/services/security_solution';
-import {
+  removeUUIDFromActions,
   getActionsWithFrequencies,
   getActionsWithoutFrequencies,
   getSomeActionsWithFrequencies,
-} from '../../utils/get_rule_actions';
-import { removeUUIDFromActions } from '../../utils/remove_uuid_from_actions';
+  updateUsername,
+} from '../utils';
+import {
+  createUserAndRole,
+  deleteUserAndRole,
+} from '../../../../common/services/security_solution';
+import { EsArchivePathBuilder } from '../../../es_archive_path_builder';
 
-// eslint-disable-next-line import/no-default-export
 export default ({ getService }: FtrProviderContext) => {
   const esArchiver = getService('esArchiver');
   const supertest = getService('supertest');
   const supertestWithoutAuth = getService('supertestWithoutAuth');
   const log = getService('log');
   const es = getService('es');
+  const config = getService('config');
+  const ELASTICSEARCH_USERNAME = config.get('servers.kibana.username');
+  const isServerless = config.get('serverless');
```

Comment:
is there a reason test file aware if it's run as serverless or stateful?

Reply:
Yes to load a specific index, because there are some differences between Serverless and ESS when it comes to index settings, write-read delay

Reply:
can we abstract this logic in a separate utility, that would just return correct path based on passed config in it?

Reply:
We will look into these points again as we agreed to merge this POC PR and tackle all the refactoring separately.